applications/views.py

```diff
@@ -36,20 +37,34 @@ def apply(request, page_url):
     menu = EventPageMenu.objects.filter(event=event)
 
     form = ApplicationForm(request.POST or None, form=form_obj)
-
     if form.is_valid():
-        form.save()
-        messages.success(request, _("Yay! Your application has been saved. You'll hear from us soon!"))
-
-        return render(
-            request,
-            "applications/apply.html",
-            {
-                "event": event,
-                "menu": menu,
-                "form_obj": form_obj,
-            },
-        )
+        try:
+            form.save()
+            messages.success(request, _("Yay! Your application has been saved. You'll hear from us soon!"))
+
+            return render(
+                request,
+                "applications/apply.html",
+                {
+                    "event": event,
+                    "menu": menu,
+                    "form_obj": form_obj,
+                },
+            )
+        except IntegrityError:
```

Comment:
Although this fixes the issue, it assumes that any `IntegrityError` is caused by this issue. Now this may be true at the moment, but things may change. So we could merge this and catch this issue at this point.

There is some validation on the form that appears to be trying to check against this same constraint, and that is where I'd prefer the fix to go; https://github.com/DjangoGirls/djangogirls/blob/main/applications/forms.py#L40